### PR TITLE
Add some performance tuning for local dev mode

### PR DESCRIPTION
**Pull request by jontg**
Suggested from
https://groups.google.com/forum/#!topic/cassandra-unit-users/HJtz2BLTV2E

### CODE REVIEW

File: cassandra.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ cluster_name: 'Test Cluster'
 #
 # If you already have a cluster with 1 token per node, and wish to migrate to 
 # multiple tokens per node, see http://wiki.apache.org/cassandra/Operations
-num_tokens: 256
+num_tokens: 32
```
Comment (cywjackson) [inline cassandra.yaml]: if it is a single node cluster, this wouldn't matter. why the change?

File: cassandra.yaml [hunk 2]
```diff
@@ -34,7 +34,7 @@ num_tokens: 256
 # of data centers to enable per-datacenter.
 # hinted_handoff_enabled: DC1,DC2
 # See http://wiki.apache.org/cassandra/HintedHandoff
-hinted_handoff_enabled: true
+hinted_handoff_enabled: false
```
Comment (cywjackson) [inline cassandra.yaml]: if it is a single node cluster, this wouldn't matter. why the change?

File: cassandra.yaml [hunk 3]
```diff
@@ -138,7 +138,7 @@ commit_failure_policy: stop
 # NOTE: if you reduce the size, you may not get you hottest keys loaded on startup.
 #
 # Default value is empty to make it "auto" (min(5% of Heap (in MB), 100MB)). Set to 0 to disable key cache.
-key_cache_size_in_mb:
+key_cache_size_in_mb: 1
```
Comment (cywjackson) [inline cassandra.yaml]: 1Mb is too little, key cache is very beneficial

File: cassandra.yaml [hunk 4]
```diff
@@ -149,7 +149,7 @@ key_cache_size_in_mb:
 # has limited use.
 #
 # Default is 14400 or 4 hours.
-key_cache_save_period: 14400
+key_cache_save_period: 0
```
Comment (cywjackson) [inline cassandra.yaml]: does 0 mean forever or no cache? no keycache =bad

File: cassandra.yaml [hunk 5]
```diff
@@ -244,8 +244,8 @@ seed_provider:
 # On the other hand, since writes are almost never IO bound, the ideal
 # number of "concurrent_writes" is dependent on the number of cores in
 # your system; (8 * number_of_cores) is a good rule of thumb.
-concurrent_reads: 32
-concurrent_writes: 32
+concurrent_reads: 2
+concurrent_writes: 2
```
Comment (cywjackson) [inline cassandra.yaml]: lower the read makes sense, since we only got 1-2 disks usually
not sure on lowering the write. should check whats the default?

File: cassandra.yaml [hunk 6]
```diff
@@ -473,7 +473,7 @@ in_memory_compaction_limit_in_mb: 64
 #
 # concurrent_compactors defaults to the number of cores.
 # Uncomment to make compaction mono-threaded, the pre-0.8 default.
-#concurrent_compactors: 1
+concurrent_compactors: 1
```
Comment (cywjackson) [inline cassandra.yaml]: this could slow down compaction (on single/test), why?

**Comment by cywjackson** [comment]
i reviewed b4 reading the link posted. reading the link now

**Comment by cywjackson** [comment]
read the post, nate is a good friend of mine (interviewed me at datastax, referred me in apigee), bit i am gonna disagree with his assessments especially on the key caches. looks like those recommendations were also done without root causing the real issue in the post later (schema change). 

and the post is old (2015).

**Comment by jontg** [comment]
Great feedback; after review I think this idea was half-baked, and I'll open a separate PR with new ideas.
